### ISSUE TITLE
Invalid condition if an abilitiy contains '!' 

### ISSUE THREAD

**Issue by cosinussinus** [axc450/pbs#51]
![par](https://github.com/axc450/pbs/assets/75318945/445ca406-3734-45e7-9eb3-fce69916a4a7)

**Comment by gizzmo** [comment]
Intersting. That exclamation mark (!) in the name is breaking the [pattering matching](https://github.com/axc450/pbs/blob/c31c6f418f7ce1bd98dbad2b43bfc916eeff6500/Core/Condition.lua#L174). Best option is to remove it. Since you have the spell id, its not required.

`use(#1) [ self.ability(BONESTORM:1761).usable ]`

**Comment by cosinussinus** [comment]
The problem is that pbs allows ability input as plain text.
'[self.ability(BONESTORM).usable]' won't work

**Comment by bloerwald** [comment]
> The problem is that pbs allows ability input as plain text.
> '[self.ability(BONESTORM).usable]' won't work

Yeah, but please never do that. When I share scripts myself, I even change the ability names to what they are semantically, like „filler“ or „buff”. Matching abilities by name has always been a bad idea.

The script with the exclamation mark should of course work though, this is a real bug. Just use the workaround until it is fixed though.
